plugins/out_s3/s3.c

```diff
@@ -1394,91 +1618,129 @@ static void cb_s3_flush(const void *data, size_t bytes,
                       ctx->timer_ms / 1000);
 
         sched = flb_sched_ctx_get();
-        ret = flb_sched_timer_cb_create(sched, FLB_SCHED_TIMER_CB_PERM,
-                                        ctx->timer_ms,
-                                        cb_s3_upload,
-                                        ctx, NULL);
+
+        if (ctx->preserve_data_ordering) {
+            ret = flb_sched_timer_cb_create(sched, FLB_SCHED_TIMER_CB_PERM,
+                                            ctx->timer_ms, s3_upload_queue, ctx, NULL);
```

Comment:
should the timer frequency be different if this feature is enabled? Normally its frequency is related to the upload_timeout.